docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,228 @@ The remaining points can happen any time after 1) is complete although note that
+     * If we really want to allow naked dummy tasks in some circumstances then
+       we would be better to add a suite configuration for this?
+     * Note: we don't allow you to inherit from an undeclared ("naked") family
+       so why do we allow you to run an undeclared task?
 
-5. Handle `rose-suite.conf` functionality
-   * Pull items from other locations / sources on installation (e.g. svn).
-   * Configure Jinja2 / EmPy inputs.
-   * Set environment variables exported to the suite daemon?
-   * Continue to handle `rose-suite.conf` using a plugin preprocessor.
-   * Decide which features to support natively in Cylc and how.
+5. Handle
+   [rose-suite.conf](https://metomi.github.io/rose/doc/html/api/configuration/suite.html#rose:file:rose-suite.conf)
+   functionality including optional configurations. See
+   [cylc-flow #3819](https://github.com/cylc/cylc-flow/issues/3819)
+   and [rose #2412](https://github.com/metomi/rose/pull/2412).
 
 6. Utility to remove installed locations (on all platforms) on demand
    (equivalent to `rose suite-clean`).
 
-7. On starting up a new run of the suite assign a new run name:
+7. Use a different run directory for each installation of a suite:
    ```
-   mi-aa001/
+   u-ab123/
      run1
      run2
    ```
-   * This is new functionality but Cylc 8 is a good time to introduce it.
+   * This is big change in working practise but Cylc 8 is a good time to
+     introduce it.
    * Use a symbolic link to identify the latest run which is the default name
      when performing operations on a suite?
 
-8. Consider the behaviour created by the `CYLC_VERSION` and `ROSE_VERSION`
-   variables in `rose-suite.conf`. These allow rose suite-run to:
-   * Change the version of Cylc and check the version of Rose used to run the
-     suite.
-   * Prevent suites accidentally upgrading or downgrading versions on restart.
-
-9. Consider `rose prune` functionality.
+8. Consider `rose prune` functionality.
    * Need to make sure it will continue to work correctly.
-   * Need Cylc equivalent but this should be done as part of
+   * Longer term we should build housekeeping into Cylc, see
      [cylc-flow #1159](https://github.com/cylc/cylc-flow/issues/1159).
 
-### CLI
+9. Record version control information when you install a suite, see
+   [cylc-flow #3849](https://github.com/cylc/cylc-flow/issues/3849).
+   * `rose suite-run` current supports this for Subversion only
+     (`svn info` + `svn status` + `svn diff`).
+   * Extend this to support equivalent information for git.
+
+### CLI changes
+
+`cylc install` - new command to install workflows.
+* By default, `cylc install` will install the workflow found in `$PWD` into
+  `~/cylc-run/$(basename $PWD)/runN` (where `runN` = `run1`, `run2` ...).
+* If `run1` already exists it will install into `run2` (and so on).
+  Create a symlink `runN` pointing at the latest run.
+* `--run-name=my-run` implies install into `~/cylc-run/$(basename $PWD)/my-run`.
```

Comment:
Possible enhancement suggestion:
`--run-name-pattern=my-run-%(n)d` implies install similar logic to default, but with specified alternative to `runN`

Reply:
Should probs make this pattern a global config where a blank value implies don't use a run dir?

Reply:
I'm not sure a global config is a good idea since it will cause `cylc install` to behave differently which makes support more difficult - they can always set up a command alias.

I'd suggest leaving any enhancement until we see whether there is demand (it wouldn't require any new command line option).

Reply:
`--run-name-pattern=road-experiment-%(n)d` I'm not sure this sort of global replacement of `run%d` would add much.  Wouldn't it be better to name the flow instead @wxtim ?  (So, `road-experiment/run1` etc.)

Reply:
> Should probs make this pattern a global config where a blank value implies don't use a run dir?

What do you mean by "don't use a run dir" @oliver-sanders ? Replicate the current model of overwriting the existing one every time?

Reply:
(Ah, `--no-run-name` just below might have answered my question)

Reply:
> I'd suggest leaving any enhancement until we see whether there is demand (it wouldn't require any new command line option).

Fair enough, happy with that, my main motive in posting it was to make it easy to disable the run dir hier.

Reply:
I'm not sure why we should allow disabling the new run1, run2, ... structure either (if that's what you mean). Is there some valid case where continually overwriting previous run-dirs is actually desirable?

Reply:
Course not, this just comes out of the fear that users might find run1, run2, ... somewhat prescriptive.

Ultimately it's just a registration pattern, it's not a hard-baked Cylc feature.